lib/rgb_matrix/animation/simon.ex

```diff
@@ -0,0 +1,134 @@
+defmodule RGBMatrix.Animation.Simon do
+  @moduledoc """
+  An interactive "Simon" game.
+  """
+
+  alias Chameleon.HSV
+  alias RGBMatrix.Animation
+
+  use Animation
+
+  defmodule Config do
+    @moduledoc false
+    use RGBMatrix.Animation.Config
+  end
+
+  defmodule State do
+    @moduledoc false
+    defstruct [:leds, :simon_sequence, :state]
+  end
+
+  @black Chameleon.RGB.new(0, 0, 0)
+  @red Chameleon.RGB.new(255, 0, 0)
+
+  @impl true
+  def new(leds, _config) do
+    state =
+      %State{leds: leds}
+      |> init_sequence()
+
+    {0, state}
+  end
+
+  @impl true
+  def render(%{state: {:playing_sequence, []}} = state, _config) do
```

Comment:
No, this is also the source of some confusion from your previous code review of the refactor (this is why the callbacks use `any` for state, not `Animation.t()`. I wanted to avoid passing in the actual animation struct in the refactor to keep things simpler (i.e. the animation implementer knows exactly what they're getting because they defined it themselves.)

This is open for discussion.

Reply:
Also, I might have contributed to that confusion here, sorry!  There's actually a field called `state` inside the `State` struct.  So starting from the animation struct it actually looks like `animation.state.state` :sweat_smile: